File: typescript/Structural/07-Bridge/Notification/UrgentNotification.ts
import { INotificationChannel } from "../Channel/INotificationChannel";
import { JobNotification } from "./JobNotification";

export class UrgentNotification extends JobNotification {
  constructor(
    channel: INotificationChannel,
    private readonly repeatCount: number = 3,
  ) {
    super(channel);
  }

  notify(jobId: string, recipient: string, message: string): void {
    for (let i = 1; i <= this.repeatCount; i++)
      this.channel.send(
        recipient,
        `🚨 URGENT (#${i}/${this.repeatCount}) Job #${jobId}`,
        message,
      );
  }
}

File: csharp/Structural/07-Bridge/Notification/UrgentNotification.cs
using Bridge.Channel;

namespace Bridge.Notification;

public class UrgentNotification : JobNotification
{
    private readonly int _repeatCount;

    public UrgentNotification(INotificationChannel channel, int repeatCount = 3) : base(channel)
    {
        _repeatCount = repeatCount;
    }

    public override void Notify(string jobId, string recipient, string message)
    {
        for (int i = 1; i <= _repeatCount; i++)
            Channel.Send(recipient, $"🚨 URGENT (#{i}/{_repeatCount}) Job #{jobId}", message);
    }
}
